ompi/mca/coll/libnbc/nbc_internal.h

```diff
@@ -565,7 +565,7 @@ static inline void NBC_SchedCache_dictwipe(hb_tree *dict_in, int *size) {
 #define NBC_IN_PLACE(sendbuf, recvbuf, inplace) \
 { \
   inplace = 0; \
-  if(recvbuf == sendbuf) { \
+  if(recvbuf == sendbuf && MPI_BOTTOM != sendbuf) { \
```

Comment:
Is this test really valid ? The MPI standard prohibits using a recvbuf equal to the sendbuf to indicate an IN_PLACE operation. Why is this macro accepting it ?

Reply:
I get your point, but are you 100% sure about this?

Bad things will definitely occurs if `sendbuf == recvbuf` **and** for example the same datatype is used.
But if the user pass `sendbuf == recvbuf` with for example two datatypes that describes non overlapping subarrays, would that violate the standard? if yes, is there an explicit exception for `MPI_BOTTOM` ?
Also, should we enhance the `C` bindings and error out with a user friendly error code if the user passes `sendbuf == recvbuf` (unless `MPI_BOTTOM`) ?

Reply:
Correct, the standard prohibit overlapping regions between the typemap of the send and receive buffers. So, while it is legal to use `sendbuf == recvbuf` it can never represent an IN_PLACE operation. So, that macro seems a little extreme in the detection of IN_PLACE.